app/queries/consumer_apps/find_or_create_by_query.rb

```diff
@@ -0,0 +1,28 @@
+module ConsumerApps
+  class FindOrCreateByQuery
+    def self.call(app_bundle:, platform:)
+      new(app_bundle: app_bundle, platform: platform).call
+    end
+
+    def initialize(app_bundle:, platform:)
+      @app_bundle = app_bundle
+      @platform = platform
+      @forem_bundle = ConsumerApp::FOREM_BUNDLE
+    end
+
+    def call
+      # This guard clause handles the Forem app special case
+      return forem_app_target if @app_bundle == @forem_bundle
+
+      # All other ConsumerApp are simply fetched as usual
+      ConsumerApp.find_by(app_bundle: @app_bundle, platform: @platform)
+    end
+
+    private
+
+    def forem_app_target
+      target = ConsumerApp.find_by(app_bundle: @forem_bundle, platform: @platform)
+      target || ConsumerApp.create(app_bundle: @forem_bundle, platform: @platform, active: true)
```

Comment:
non blocking:

It's unlikely but is technically a race condition. If between `.find_by` and `.create` a second admin creates the consumer app, this will fail. 

You could use [create_or_find_by](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-create_or_find_by) which is a little better than `find_or_create_by` but it requires a unique constraint (which I see we've added :D)

Reply:
Very unlikely indeed, but worth making sure this edge case is handled as good as possible rather than wait until it causes a problem in the future. Done ✅